qiskit/validation/fields/custom.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Fields custom to Terra to be used with Qiskit validated classes."""
+
+import numpy
+import sympy
+
+from marshmallow.utils import is_collection
+
+from qiskit.validation import ModelTypeValidator
+
+
+class Complex(ModelTypeValidator):
```

Comment:
do we still need this class, given that the more general `InstructionParameter` class covers the case of `complex`?

Reply:
Yes, as it is the one used in `Result` and other places where the specs specify that a complex is expected - this way we can be as specific as we want there.